### PR TITLE
demote trunk from divided highways to normal roads, anounce ramps whe…

### PR DESCRIPTION
**Pull request by MoKob**
…n entering a motorway directly

# Issue

resolves https://github.com/Project-OSRM/osrm-backend/issues/4274

## Tasklist
 - [x] add regression / cucumber cases (see docs/testing.md)
 - [ ] review
 - [ ] adjust for comments

### CODE REVIEW

**Comment by TheMarex** [comment]
How does this affect scenarios that use a `trunk_link` like [this example](http://map.project-osrm.org/?z=17&center=49.016236%2C8.350881&loc=49.014061%2C8.351712&loc=49.018593%2C8.348193&hl=en&alt=0)? Do we emit a `on ramp` in this case, or would this become a `turn right`?

**Comment by MoKob** [comment]
Sadly these would be labeled a simple turn. The problem is that we need a way to find out about these dual-carriage ways. We might be able to assign the motorway tag based on available references/country level.
While the linked example would surely be best described as a ramp, we have many other cases just like [here](https://www.openstreetmap.org/way/8922554) [osrm](http://map.project-osrm.org/?z=18&center=37.800640%2C-122.425488&loc=37.800968%2C-122.426577&loc=37.800052%2C-122.424302&hl=en&alt=0) which surely aren't ramps :(

**Comment by TheMarex** [comment]
I think the OSM tagging in the highway `101` case is actually wrong. This segment is clearly part of the normal traffic flow of the highway. For example [this](https://www.openstreetmap.org/way/23876482) segment also just connects two `trunk` roads but is also tagged `trunk`.

We should first get a good understanding of what the trade-off here is. We might be improving instructions inside some US cities but seems like we could vastly decrease our guidance quality in other areas. Maybe the answer here can be to insist on better tagging on OSM side.

**Comment by MoKob** [comment]
I think you are right and we should investigate further. It might be that we just need another level of country awareness, based on tagging schemes. I am tagging this as experimental and feature request for now.

File: src/extractor/guidance/motorway_handler.cpp [hunk 1]
```diff
@@ -62,7 +62,7 @@ bool MotorwayHandler::canProcess(const NodeID,
     for (const auto &road : intersection)
     {
         // not merging or forking?
-        if (road.entry_allowed && angularDeviation(road.angle, STRAIGHT_ANGLE) > 60)
+        if (road.entry_allowed && angularDeviation(road.angle, STRAIGHT_ANGLE) > 91)
```
Comment (daniel-j-h) [inline src/extractor/guidance/motorway_handler.cpp]: Why 91?!

**Pull request closed without merging**